### PR TITLE
feat: make ProxyAdmin self-owned Proxy

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Turns the ProxyAdmin into a Proxy that owns itself. Result of this is that we can treat the ProxyAdmin as a Proxy which reduces some complexity in the migration. We can then also upgrade the ProxyAdmin on L2.

### CODE REVIEW

**Comment by tynes** [comment]
The `check-l2` script needs to be updated to check for the admin + impl slots being set at the `ProxyAdmin` predeploy account

**Comment by mslipper** [comment]
Confirming - the owner of the `ProxyAdmin` would be the multisig. The admin of the proxy in front of `ProxyAdmin` would be itself. This would let us use `ProxyAdmin` to self-upgrade by calling `ProxyAdmin` from the multisig. Is this correct?

**Comment by smartcontracts** [comment]
> Confirming - the owner of the `ProxyAdmin` would be the multisig. The admin of the proxy in front of `ProxyAdmin` would be itself. This would let us use `ProxyAdmin` to self-upgrade by calling `ProxyAdmin` from the multisig. Is this correct?

That is correct.

**Comment by mslipper** [comment]
This looks good to me. Would like a review from @tynes or @maurelian before merging.

**Comment by mslipper** [comment]
Closing in favor of an alternate proposal.

File: packages/contracts-bedrock/tasks/check-l2.ts
Comment (maurelian) [inline packages/contracts-bedrock/tasks/check-l2.ts:582]:
```suggestion
    await checkProxy(hre, 'ProxyAdmin', signer.provider)
    await assertProxy(hre, 'ProxyAdmin', signer.provider)
```

Comment (tynes) [reply packages/contracts-bedrock/tasks/check-l2.ts:582]: ^^ This is correct, will break when using a URL passed via CLI if not

File: op-chain-ops/genesis/setters.go
Comment (tynes) [inline op-chain-ops/genesis/setters.go:175]: nit: unnecessary diff

File: op-chain-ops/genesis/layer_two_test.go
Comment (tynes) [inline op-chain-ops/genesis/layer_two_test.go:95]: There is now 1 more additional code address in the state

File: packages/contracts-bedrock/tasks/check-l2.ts
Comment (tynes) [reply packages/contracts-bedrock/tasks/check-l2.ts:582]: CI failing with this error:

```
tasks/check-l2.ts(581,11): error TS2554: Expected 3 arguments, but got 2.
```

**Comment by maurelian** [comment]
Specs need a small change too: 

https://github.com/ethereum-optimism/optimism/blob/9a2ffd119e18a49e0264edcd29c6c6a6206755e1/specs/predeploys.md#L256-L257

**Review by maurelian — APPROVED** [review]

**Comment by mslipper** [comment]
Running against a datadir now. Will approve once I see that work.

File: op-chain-ops/genesis/check.go
Comment (mslipper) [reply op-chain-ops/genesis/check.go:78]: Could've sworn I'd fixed this as part of another PR - checking

Comment (mslipper) [reply op-chain-ops/genesis/check.go:78]: Ah, they weren't fully fixed there either. Fixing fully now.

**Comment by mslipper** [comment]
Migration against the datadir worked. Merging, and will perform an end-to-end migration tonight.

**Review by mslipper — APPROVED** [review]

**Pull request merged**
